community_images/nats/bitnami/functional_test.sh

```diff
@@ -0,0 +1,110 @@
+#!/bin/bash
+
+set -x
+set -e
+
+SCRIPTPATH="$( cd -- "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
+
+# shellcheck disable=SC1091
+. "${SCRIPTPATH}"/../../common/helpers.sh
+
+HELM_RELEASE=rf-nats
+NAMESPACE=$(get_namespace_string "${HELM_RELEASE}")
+REPOSITORY=nats
+IMAGE_REPOSITORY="$RAPIDFORT_ACCOUNT"/"$REPOSITORY"
+
+k8s_test()
+{
+    # setup namespace
+    setup_namespace "${NAMESPACE}"
+
+    # install helm
+    with_backoff helm install "${HELM_RELEASE}" bitnami/"$REPOSITORY" --set image.repository="$IMAGE_REPOSITORY" --set image.tag=latest --namespace "${NAMESPACE}"
+    report_pulls "${IMAGE_REPOSITORY}"
+
+    # wait for pods
+    kubectl wait pods "${HELM_RELEASE}"-0 -n "${NAMESPACE}" --for=condition=ready --timeout=10m
+
+    NATS_USER=$(kubectl get secret --namespace ${NAMESPACE} ${HELM_RELEASE} -o jsonpath='{.data.*}' | base64 -d | grep -m 1 user | awk '{print $2}' | tr -d '"')
+    NATS_PASS=$(kubectl get secret --namespace "${NAMESPACE}" "${HELM_RELEASE}" -o jsonpath='{.data.*}' | base64 -d | grep -m 1 password | awk '{print $2}' | tr -d '"')
+    echo -e "Client credentials:\n\tUser: $NATS_USER\n\tPassword: $NATS_PASS"
+
+    kubectl run nats-release-client --restart='Never' --env="NATS_USER=$NATS_USER" --env="NATS_PASS=$NATS_PASS" --image docker.io/bitnami/golang --namespace "${NAMESPACE}" --command -- sleep infinity
+    kubectl wait pods nats-release-client -n "${NAMESPACE}" --for=condition=ready --timeout=10m
+    echo "#!/bin/bash
+    GO111MODULE=off go get github.com/nats-io/nats.go
```

Comment:
is this same as commands.sh? Do we need to remove this section? Also can we make a function?

Reply:
yes, there are few dynamic values going into the commands.sh file, i have added a line to delete commands.sh once it's copied to container so commands.sh file is not going to be checked into the code